File: java/Hardware.java
class Hardware {
    protected String name, brand;
    protected double price;
    protected int warrantyPeriod;
    public Hardware(String name, double price, int warrantyPeriod, String brand) {
        this.name = name;
        this.price = price;
        this.warrantyPeriod = warrantyPeriod;
        this.brand = brand;
    }
    public String getName() { return name; }
    public String getBrand() { return brand; }
    public double getPrice() { return price; }
    public int getWarrantyPeriod() { return warrantyPeriod; }
    // setter
    public void setName(String name) { this.name = name; }
    public void setBrand(String brand) { this.brand = brand; }
    public void setPrice(double price) { this.price = price; }
    public void setWarrantyPeriod(int warrantyPeriod) { this.warrantyPeriod = warrantyPeriod; }

    
}

File: cpp/Hardware.cpp
#pragma once
#include <iostream>
#include <vector>
using namespace std;

class Hardware {
    protected:
        string name;
        float price;
        int warrantyPeriod; // dalam tahun
        string brand;
    
    public:
        Hardware(string n, float p, int w, string b) 
            : name(n), price(p), warrantyPeriod(w), brand(b) {}
    
        // Getters
        string getName() const { return name; } // Add const
        double getPrice() const { return price; }
        int getWarrantyPeriod() const { return warrantyPeriod; }
        string getBrand() const { return brand; }

    
        // Setters
        void setName(string n) { name = n; }
        void setPrice(float p) { price = p; }
        void setWarrantyPeriod(int w) { warrantyPeriod = w; }
        void setBrand(string b) { brand = b; }
    };